python/tvm/relax/op/gradient.py

```diff
@@ -0,0 +1,194 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Gradient definitions for Relax operators"""
+from tvm.relax.op import register_gradient
+from tvm.relax.op import sum as _sum
+from tvm.relax.op import (
+    mean,
+    less,
+    where,
+    collapse_sum_to,
+    broadcast_to,
+    multiply,
+    negative,
+    subtract,
+    permute_dims,
+    matmul,
+    ones,
+    zeros,
+    expand_dims,
+)
+from tvm.relax.expr import Call, Var
+
+
+@register_gradient("relax.add")
+def add_grad(orig: Call, grad: Var):
+    """Returns [grad, grad]."""
+    return [
+        collapse_sum_to(grad, orig.args[0].struct_info.shape),
+        collapse_sum_to(grad, orig.args[1].struct_info.shape),
+    ]
+
+
+@register_gradient("relax.subtract")
+def subtract_grad(orig: Call, grad: Var):
+    """Returns [grad, -grad]."""
+    return [
+        collapse_sum_to(grad, orig.args[0].struct_info.shape),
+        collapse_sum_to(negative(grad), orig.args[1].struct_info.shape),
+    ]
+
+
+@register_gradient("relax.multiply")
+def multiply_grad(orig: Call, grad: Var):
+    """Returns [grad * y, grad * x]."""
+    x, y = orig.args
+    return [
+        collapse_sum_to(multiply(grad, y), x.struct_info.shape),
+        collapse_sum_to(multiply(grad, x), y.struct_info.shape),
+    ]
+
+
+@register_gradient("relax.permute_dims")
+def permute_dims_grad(orig: Call, grad: Var):
+    """Returns grad transposed over the complement of original permute_dims axes."""
+    axes = orig.attrs["axes"]
+    if axes:
```

Comment:
Fixed, just for record

We should use the **inverse** of `axes`, instead of `axes`, as the new axes of the gradient of permute_dims.